Bootstrap/Helpers.ts

```diff
@@ -1,46 +1,44 @@
+import { isLinux } from "../Library/PrefixHelper";
 import { DiagnosticLog, DiagnosticMessageId } from "./DataModel";
 import { DiagnosticLogger } from "./DiagnosticLogger";
 
-const USER_APP_PATH = "/home/site/wwwroot";
+const LINUX_USER_APPLICATION_INSIGHTS_PATH = "/node_modules/applicationinsights/out/applicationinsights.js";
 
 export function sdkAlreadyExists(_logger: DiagnosticLogger): boolean {
     try {
         // appInstance should either resolve to user SDK or crash. If it resolves to attach SDK, user probably modified their NODE_PATH
         let appInstance: string;
         try {
             // Node 8.9+
-            // If we find the applicationinsights module under in the user application path, do not attach the SDK
-            // In order for this to work in Windows, we need to pass the full "/home/site/wwwroot" path to require.resolve
-            appInstance = (require.resolve as any)("applicationinsights", { paths: [USER_APP_PATH] });
-            if (appInstance) {
-                diagnosticLogSdkExists(_logger, appInstance);
-                return true;
-            }
+            appInstance = (require.resolve as any)("applicationinsights", { paths: [process.cwd()] });
         } catch (e) {
             // Node <8.9
             appInstance = require.resolve(process.cwd() + "/node_modules/applicationinsights");
-            // If loaded instance is in Azure machine home path do not attach the SDK, this means customer already instrumented their app
-            if (appInstance.indexOf("home") > -1) {
-                diagnosticLogSdkExists(_logger, appInstance);
-                return true;
-            }
-            else {
-                // ApplicationInsights could be loaded outside of customer application, attach in this case
-                return false;
-            }
+        }
+        /** 
+         * If loaded instance is in Azure machine home path do not attach the SDK, this means customer already instrumented their app.
+         * Linux App Service doesn't append the full cwd to the require.resolve, so we need to check for the relative path we expect
+         * if application insights is being imported in the user app code.
+        */
+        if (
+            appInstance.indexOf("home") > -1 ||
+            (appInstance === LINUX_USER_APPLICATION_INSIGHTS_PATH && isLinux())
```

Comment:
I don't fully understand this check, in normal circumstances when ai is installed in customer app,is not app instance part of home path?

Reply:
It is under the `home` path, but `require.resolve()` behaves differently in Linux vs. Windows app services. in Windows it returns the full path, something like `/home/site/wwwroot/<path-to-user-instance-of-app-insights-package>` but in Linux it returns just the path starting from under `wwwroot`, which is why our original check wouldn't back off in cases where the customer had instrumented their code.

This new check is just aware of the fact that Linux behaves that way and knows that the `/node_modules/applicationinsights/out/applicationinsights.js` path is always from under `/home`. In cases where `require.resolve()` recognizes the other agent in the app service it'll be under some `/usr/...` path, which won't stop our attach.